clients/src/main/java/org/apache/kafka/clients/admin/DescribeTopicsResult.java

```diff
@@ -31,29 +31,29 @@
  * The API of this class is evolving, see {@link Admin} for details.
  */
 @InterfaceStability.Evolving
-public class DescribeTopicsResult {
-    private final Map<String, KafkaFuture<TopicDescription>> futures;
+public class DescribeTopicsResult<T> {
```

Comment:
I think this parameterization is a pretty good idea, and I can add it to my delete topics PR. But if we are going to change the public API, we should update the KIP and potentially update the mailing list with the changes.

Reply:
I'm also curious what implications there are for changing the original DescribeTopicsResult when using old clients.

Reply:
I first tried a new class `DescribeTopicsResultWithIds` and find it's almost the same as `DescribeTopicsResult`, so I just use parameterization. We only need to change `DescribeTopicsResult` to `DescribeTopicsResult<String>` when using old clients.

Reply:
We need to keep the Admin API backwards compatible. An application that was written using the 2.7.0 should not break if it is compiled with a 2.8.0 clients jar. You can always add an internal class with shared code to avoid duplication, but the public API itself needs to remain compatible.

Reply:
Thank you for your reminder, I have rollbacked all the public API change.